experimental/uefi/loader/src/main.rs

```diff
@@ -79,34 +78,53 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         }
     });
 
-    // Bit hacky, but it's only temporary and turns out console I/O is complex.
-    let (mut rh, mut wh) = comms.into_split();
-    tokio::spawn(async move {
-        let mut buf = [0; 1024];
-        loop {
-            let result = rh.read(&mut buf).await.unwrap();
-            if result == 0 {
-                break;
-            } else {
-                info!("rx: {:?}", &buf[..result]);
-            }
-        }
-    });
+    // Unfortunately OVMF writes some garbage (clear screen etc?) + our Hello World to the
+    // other serial port, so let's skip some bytes before we set up the CBOR codec.
+    let mut junk = [0; 71];
+    let mut len = 0;
+    while len < junk.len() {
+        len += comms.read(&mut junk[len..]).await.unwrap();
+    }
+    info!("Leading junk on comms: {:?}", std::str::from_utf8(&junk));
+
+    // Use a bmrng channel for serial communication. The good thing about spawning a separate
+    // task for the serial communication is that it serializes (no pun intended) the communication,
+    // as right now we don't have any mechanisms to track multiple requests in flight.
+    let (tx, mut rx) = bmrng::unbounded_channel::<String, anyhow::Result<String>>();
+
     tokio::spawn(async move {
-        let mut buf = [0; 1024];
-        loop {
-            let result = io::stdin().read(&mut buf).await.unwrap();
-            if result == 0 {
-                break;
-            } else {
-                info!("tx: {:?}", &buf[..result]);
-                wh.write_all(&buf[..result]).await.unwrap();
-            }
+        // Set up the CBOR codec to handle the comms.
+        let codec: Codec<String, String> = Codec::new();
+        let mut channel = codec.framed(comms);
+        while let Ok((input, responder)) = rx.recv().await {
+            responder
+                .respond({
+                    if let Err(err) = channel.send(input).await {
+                        Err(anyhow::Error::from(err))
+                    } else {
+                        // Sometimes next() gives us a None. Figure out what's going on in there.
```

Comment:
This seems like a hidden TODO. Perhaps create an issue and add an explicit TODO here.